domain_models/views.py

```diff
@@ -0,0 +1,30 @@
+"""Contextual view module."""
+from domain_models import models
+
+
+class ContextView(object):
+    """Contextual view class."""
+
+    __model__ = None
+
+    def __init__(self, model):
+        """Model validation.
+
+        :param model: DomainModel
+        """
+        if not issubclass(self.__model__, models.DomainModel):
+            raise TypeError("Attribute __model__ must be subclass of "
+                            "DomainModel")
+
+        if not isinstance(model, self.__model__):
+            raise TypeError("\"{0}\" is not an instance of {1}".format(
+                model, self.__model__))
+
+        self.__model__ = model
```

Comment:
I think that it will be cleaner of we would leave ```__model__``` attribute as read-only class attribute, and set model instance into something like ```model_object```, or ```__model__``` -> ```__model_cls__``` and ```self.model = model```.